### PR TITLE
feat: push chat integration

### PR DESCRIPTION
**Pull request by b4s36t4**
## Related issues

Fixes https://github.com/heyxyz/hey/issues/4240
/claim #4240

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Comment by b4s36t4** [comment]
@bigint bump to review/check the progress.

**Comment by b4s36t4** [comment]
https://github.com/heyxyz/hey/assets/59088937/3abc09f7-5465-4643-9969-365269d2501a


cc: @5war00p for his side view.

**Comment by 5war00p** [comment]
https://github.com/heyxyz/hey/assets/57614947/db37e126-6169-4913-b227-8d38b1e6a23a

This shows approval and rejection of a request and the messages after acceptance. Also code is not refined yet, please don't review deeply will refine it later.

**Comment by bigint** [comment]
Cool are you guys both collaborating?

File: apps/web/package.json [hunk 1]
```diff
@@ -33,6 +33,8 @@
     "@lexical/markdown": "^0.12.5",
     "@lexical/react": "^0.12.5",
     "@lexical/utils": "^0.12.5",
+    "@pushprotocol/restapi": "^1.4.45",
```
Comment (bigint) [inline apps/web/package.json:36]: why not their react sdk?

**Comment by bigint** [comment]
we gonna need to store the `privatepgpkey` on localstorage as well!

**Comment by 5war00p** [comment]
> Cool are you guys both collaborating?

Yes we are.

File: apps/web/package.json (hunk 1 above)
Comment (b4s36t4) [reply apps/web/package.json:36]: It is their react-sdk, they have ui-components, nothing specific react sdk. Even their react example is also using same method.

**Comment by b4s36t4** [comment]
> Cool are you guys both collaborating?

Yes, we are :)

**Comment by b4s36t4** [comment]
@bigint can you share more info on `privatepgpkey`?

**Comment by bigint** [comment]
@b4s36t4 so when you sign on the push for the first time it will emit private pgp key, we need to store it and use on further API calls, so user don't need to sign everytime!

File: apps/web/package.json (hunk 1 above)
Comment (bigint) [reply apps/web/package.json:36]: Ohh you are correct!

**Comment by bigint** [comment]
Screenshot 2023-12-17 at 7 53 29 PM

Preview broke for me!

**Comment by 5war00p** [comment]
@bigint Look into this: https://github.com/ethereum-push-notification-service/push-sdk/blob/main/packages/restapi/src/lib/pushapi/pushAPITypes.ts#L8

The initialize function consider only these options, there is no way to pass signer.

File: apps/web/src/store/persisted/usePushStore.ts
Comment (bigint) [inline apps/web/src/store/persisted/usePushStore.ts:1]: Rename to useMessageStore

File: apps/web/src/pages/messages/index.tsx [hunk 2]
```diff
@@ -1,45 +1,55 @@
...
-import { useState } from 'react';
 import useProfileStore from 'src/store/persisted/useProfileStore';
+import usePushStore from 'src/store/persisted/usePushStore';
 import { useUpdateEffect } from 'usehooks-ts';
 import { useWalletClient } from 'wagmi';
 
 const ChatPage = () => {
   const { data: signer, isLoading } = useWalletClient();
-
+  const { pgpPvtKey, setPgpPvtKey } = usePushStore();
```
Comment (bigint) [inline apps/web/src/pages/messages/index.tsx:13]: always select individually

`const pgpPvtKey = usePushStore((state) => state.pgpPvtKey);`

**Review by bigint — CHANGES_REQUESTED** [review]

File: apps/web/src/pages/messages/index.tsx (hunk 2 above)
Comment (b4s36t4) [reply apps/web/src/pages/messages/index.tsx:13]: @5war00p  please check these.

**Comment by bigint** [comment]
Are we still on track?

**Comment by 5war00p** [comment]
@bigint  Yes, emoji reactions is done. you can test that out.

**Comment by 5war00p** [comment]
Next things are replies, attachments.
What else are there ?

**Comment by bigint** [comment]
stuck here 🙈

**Comment by 5war00p** [comment]
This is one thing we need look into, unlock your metamask extension.

**Comment by bigint** [comment]
FYI there is another PR 🙇🏼 #4321

**Comment by bigint** [comment]
Frens any updates? we have a deadline of jan 10 🙇🏼

**Comment by bigint** [comment]
PR will be closed if there is no updates for another 3 days 🙇🏼

**Comment by 5war00p** [comment]
> PR will be closed if there is no updates for another 3 days 🙇🏼

Sure.

**Comment by hashtagobi1** [comment]
Hi @b4s36t4 & @5war00p,

I opened a PR yesterday to solve this but you guys are almost complete.

I am available to help out on any remaining tasks to help reach the deadline on Jan 10th.

**Comment by 5war00p** [comment]
@bigint we've pushed streaming and updated you on telegram? Did u checked?

**Comment by 5war00p** [comment]
@bigint Can you give updates or availability to check this?

**Comment by bigint** [comment]
Closing this as it didn't match the requirement, TBH it needs to be done internally rather than bounty for such big features!

**Pull request closed without merging**